six/modules/c++/cphd/include/cphd/FileHeader.h

```diff
@@ -28,122 +27,104 @@
 #include <utility>
 
 #include <io/SeekableStreams.h>
+#include <cphd/BaseFileHeader.h>
 
 namespace cphd
 {
-class FileHeader
+class FileHeader : public BaseFileHeader
 {
 public:
-    static const char FILE_TYPE[];
     static const char DEFAULT_VERSION[];
-    static const char KVP_DELIMITER[];
-    static const char LINE_TERMINATOR;
-    static const char SECTION_TERMINATOR;
-    static const size_t MAX_HEADER_SIZE;
 
     FileHeader();
 
-    static
-    bool isCPHD(io::SeekableInputStream& inStream);
-    
-    static
-    std::string readVersion(io::SeekableInputStream& inStream);
+    virtual ~FileHeader()
+    {
+    }
 
-    void read(io::SeekableInputStream& inStream);
+    virtual void read(io::SeekableInputStream& inStream);
```

Comment:
Since we have C++11 now, the better way to do this is `void read(io::SeekableInputStream& inStream) override;` (so you get compile errors if you mess up the signature)